src/test/database/remove-remote.spec.ts

```diff
@@ -0,0 +1,85 @@
+"use strict";
+
+import { expect } from "chai";
+import { SinonSandbox } from "sinon";
+import * as sinon from "sinon";
+import * as nock from "nock";
+import utils = require("../../utils");
+import api = require("../../api");
+
+import helpers = require("../helpers");
+import RemoveRemote from "../../database/remove-remote";
+
+describe("RemoveRemote", () => {
+  const instance = "fake-db";
+  const remote = new RemoveRemote(instance);
+  const serverUrl = utils.addSubdomain(api.realtimeOrigin, instance);
+  let sandbox: SinonSandbox;
+
+  beforeEach(() => {
+    sandbox = sinon.createSandbox();
+    helpers.mockAuth(sandbox);
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+    nock.cleanAll();
+  });
+
+  it("listPath should work", (done) => {
+    nock(serverUrl)
+      .get("/.json")
+      .query({ shallow: true, limitToFirst: "50000" })
+      .reply(200, {
+        a: true,
+        x: true,
+        f: true,
+      });
+    expect(remote.listPath("/"))
```

Comment:
If you sync to master (the give away will be that `mocha.opts` in the root contains `--require ... mocha-bootstrap.js`), it includes `chai-as-promised` by default, which makes dealing with promises much, much easier.

If you don't want to write this using `async`/`await`, which is fine, you can do this to save you a bit of pain

```typescript
return expect(remote.listPath("/")).to.eventually.deep.equal(["a", "x", "f"]);
```

`chai-as-promised` lets you return a promise that represents the test (which `expect`...`eventually` is). You then can remove `done` from the function signature.

This applies throughout.